### ISSUE TITLE
Accessibility attributes on search modal

### ISSUE THREAD

**Issue by swanrj08** [tomusborne/generatepress#662]
The modal is missing the necessary ARIA attributes to inform screen reader users that it is a dialog, and it also doesn't trap focus for screen reader users navigating with arrow keys. Role="dialog" and aria-modal="true" on the container for the modal to fix both these issues.

There should also be an aria-label or aria-labelledby attribute that names the modal.
